kusto/kql/identifier.go

```diff
@@ -0,0 +1,45 @@
+package kql
+
+import "fmt"
+
+type Identifier struct {
```

Comment:
Yochai doesn't think this should be a struct.
Simple:
delete the struct.
For add database, just do `return b.addBase(fmt.Sprintf("%s(%s)", "database", QuoteString(database, false)))`

For the rest:
make normalize name into a function accepting name (we don't need the wrapping anymore), and change all to b.addBase(table) etc

Reply:
i do need though something that implements String (as addBase expects fmt.Stringer).  should i deliver the values the same way as i do in AddString perhaps (with the normalize)?  
b.addBase(newValue(NormalizeName(value), types.String))